tensorflow/lite/kernels/slice.cc

```diff
@@ -121,6 +120,9 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
   TF_LITE_ENSURE_EQ(context, NumDimensions(size), 1);
   TF_LITE_ENSURE_MSG(context, NumDimensions(input) <= kMaxDim,
                      "Slice op only supports 1D-4D input arrays.");
```

Comment:
@jdduke : Sorry for confusion, actually Slice Kernel has 2 things to handle.
  1:> Support arbitrary dimension Input.
  2:> Compulsory conversion of all type of Dims input to 4-D.
So in my current PR, i have addressed the 2nd issue, where i made the kernels independent of Input Dimensions, now it can operate on any N-Dims Input.

Once this PR gets approved, i will resolve the 1st issue as well.